kong/plugins/key-auth/handler.lua

```diff
@@ -154,14 +159,19 @@ local function do_authentication(conf)
   local cache = kong.cache
 
   local credential_cache_key = kong.db.keyauth_credentials:cache_key(key)
-  local credential, err = cache:get(credential_cache_key, nil, load_credential,
+  -- hit_level be 1 if stale value is propelled into L1 cache; so set a minimal `resurrect_ttl`
+  local credential, err, hit_level = cache:get(credential_cache_key, { resurrect_ttl = 0.001 }, load_credential,
```

Comment:
We handle hit level below, so should we have this without `{ resurrect_ttl = 0.001 }`

Reply:
We should have it.

Though we have `hit_level` but sometimes expired key will be cached in L1 cache and the `hit_level` is 1 then.

So, adding `{ resurrect_ttl = 0.001 }` to overcome that case.